src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py

```diff
@@ -71,21 +69,27 @@ def _main(settings, optlist, args):
                             test_count = int(match.groups()[0])
                         break
 
-            result_url = test_executor.upload_file_to_storage(results_location, settings)
-            log.info("Sending completion event")
-            event_client.send(
-                {
-                    'Type': 'XUnitTestResult',
-                    'WorkItemId': settings.workitem_id,
-                    'WorkItemFriendlyName': settings.workitem_friendly_name,
-                    'CorrelationId': settings.correlation_id,
-                    'ResultsXmlUri': result_url,
-                    'TestCount': test_count,
-                }
+            if settings.output_uri is not None:
+                result_url = test_executor.upload_file_to_storage(results_location, settings)
+            else:
+                result_url = None;
+
+            if (settings.event_uri is not None):
```

Comment:
this should be gated on whether event_client is null and not create one earlier if the event_uri is not available.

Reply:
What do you think about moving the creation of event_client to the bottom and so we only need one condition?

```
            if (settings.event_uri is not None):
                event_client = helix.event.create_from_uri(settings.event_uri);
                log.info("Sending completion event")
                event_client.send(
                    {
                        'Type': 'XUnitTestResult',
                        'WorkItemId': settings.workitem_id,
                        'WorkItemFriendlyName': settings.workitem_friendly_name,
                        'CorrelationId': settings.correlation_id,
                        'ResultsXmlUri': result_url,
                        'TestCount': test_count,
                    }
                )
```

Reply:
totally fine.